usersync/ejector.go

```diff
@@ -0,0 +1,145 @@
+		p.OldestEjector.EjectPriority = false
+		return p.OldestEjector.Choose(uids)
+	}
+
+	if isSyncerPriority(p.SyncerKey, p.PriorityGroups) {
+		lowestPriorityGroup := p.PriorityGroups[len(p.PriorityGroups)-1]
+
+		if len(lowestPriorityGroup) == 1 {
+			uidToDelete := lowestPriorityGroup[0]
+			p.PriorityGroups = removeElementFromPriorityGroup(p.PriorityGroups, uidToDelete)
+			return uidToDelete, nil
+		} else {
+			p.OldestEjector.EjectPriority = true
+			p.OldestEjector.PriorityGroups = p.PriorityGroups
+			uidToDelete, err := p.OldestEjector.Choose(uids)
+			if err != nil {
+				return "", err
+			}
+			p.PriorityGroups = removeElementFromPriorityGroup(p.PriorityGroups, uidToDelete)
+			return uidToDelete, nil
+		}
+	}
+	return "", errors.New("syncer key " + p.SyncerKey + " is not in priority groups")
+}
+
+// updatePriorityGroup will remove the selected element from the priority groups, and will remove the entire priority group if it's empty
+func removeElementFromPriorityGroup(priorityGroups [][]string, oldestElement string) [][]string {
+	lowestPriorityGroup := priorityGroups[len(priorityGroups)-1]
+	if len(lowestPriorityGroup) <= 1 {
+		priorityGroups = priorityGroups[:len(priorityGroups)-1]
+		return priorityGroups
+	}
+
+	for index, elem := range lowestPriorityGroup {
+		if elem == oldestElement {
+			updatedPriorityGroup := append(lowestPriorityGroup[:index], lowestPriorityGroup[index+1:]...)
+			priorityGroups[len(priorityGroups)-1] = updatedPriorityGroup
+			return priorityGroups
+		}
+	}
+	return priorityGroups
+}
+
+func isSyncerPriority(syncer string, priorityGroups [][]string) bool {
+	for _, group := range priorityGroups {
+		for _, bidder := range group {
+			if syncer == bidder {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func getNonPriorityKeys(uids map[string]UIDEntry, priorityGroups [][]string) []string {
+	// If no priority groups, then all keys in uids are non-priority
+	nonPriorityKeys := []string{}
+	if len(priorityGroups) == 0 {
+		for key := range uids {
+			nonPriorityKeys = append(nonPriorityKeys, key)
+		}
+		return nonPriorityKeys
+	}
+
+	// Create map of keys that are a priority
+	isPriority := make(map[string]bool)
+	for _, group := range priorityGroups {
+		for _, bidder := range group {
+			isPriority[bidder] = true
+		}
+	}
+
+	// Loop over uids and compare the keys in this map to the keys in the isPriority map to find the non piority keys
+	for key := range uids {
+		if !isPriority[key] {
+			nonPriorityKeys = append(nonPriorityKeys, key)
+		}
+	}
+
+	return nonPriorityKeys
+}
+
+func getOldestElement(elements []string, uids map[string]UIDEntry) string {
+	var oldestElem string = ""
+	var oldestDate int64 = math.MaxInt64
+
+	for _, key := range elements {
+		if value, ok := uids[key]; ok {
+			timeUntilExpiration := time.Until(value.Expires)
+			if timeUntilExpiration < time.Duration(oldestDate) {
+				oldestElem = key
+				oldestDate = int64(timeUntilExpiration)
+			}
+		} else {
+			continue
```

Comment:
Nitpick: This `else` can be removed since it's inconsequential. Without it, the loop would just continue anyway.